templates/pytorch/nightly/fs-transformer.libsonnet

```diff
@@ -15,40 +15,55 @@
 local base = import "base.libsonnet";
 local timeouts = import "../../timeouts.libsonnet";
 local tpus = import "../../tpus.libsonnet";
+local utils = import "../../utils.libsonnet";
 
 {
   local transformer = base.PyTorchTest {
     modelName: "fs-transformer",
-    command: [
-      "python3",
-      "/tpu-examples/deps/fairseq/train.py",
-      "/datasets/wmt18_en_de_bpej32k",
-      "--tensorboard-logdir=$(MODEL_DIR)",
-      "--metrics_debug",
-      "--arch=transformer_vaswani_wmt_en_de_big",
-      "--max-target-positions=64",
-      "--attention-dropout=0.1",
-      "--no-progress-bar",
-      "--no-save",
-      "--save-interval=1",
-      "--criterion=label_smoothed_cross_entropy",
-      "--source-lang=en",
-      "--lr-scheduler=inverse_sqrt",
-      "--min-lr=1e-09",
-      "--skip-invalid-size-inputs-valid-test",
-      "--target-lang=de",
-      "--label-smoothing=0.1",
-      "--update-freq=1",
-      "--optimizer=adam",
-      "--adam-betas=(0.9,0.98)",
-      "--warmup-init-lr=1e-07",
-      "--lr=0.0005",
-      "--warmup-updates=4000",
-      "--share-all-embeddings",
-      "--dropout=0.3",
-      "--weight-decay=0.0",
-      "--valid-subset=valid",
-      "--num_cores=8",
+    paramsOverride: {
+      maxEpoch: error "Must set `maxEpoch`",
+      logSteps: error "Must set `logSteps`",
+      trainSubset: error "Must set `trainSubset`",
+      validSubset: error "Must set `validSubset`",
+      inputShapes: error "Must set `inputShapes`",
+    },
+    command: utils.scriptCommand(
+      |||
+        pip install --editable tpu-examples/deps/fairseq
+        python3 \
+              /tpu-examples/deps/fairseq/train.py \
+              /datasets/wmt18_en_de_bpej32k  \
+              --tensorboard-logdir=$(MODEL_DIR)  \
+              --metrics_debug \
+              --arch=transformer_vaswani_wmt_en_de_big \
+              --max-target-positions=64  \
+              --attention-dropout=0.1 \
+              --no-progress-bar  \
+              --save-interval=5  \
+              --save-dir=/tmp/checkpoints \
+              --criterion=label_smoothed_cross_entropy \
+              --source-lang=en \
+              --lr-scheduler=inverse_sqrt \
+              --min-lr=1e-09 \
+              --skip-invalid-size-inputs-valid-test  \
+              --target-lang=de \
+              --label-smoothing=0.1  \
+              --update-freq=1 \
+              --optimizer=adam \
+              --adam-betas=(0.9  0.98) \
+              --warmup-init-lr=1e-07 \
+              --lr=0.0005 \
+              --warmup-updates=4000  \
+              --share-all-embeddings \
+              --dropout=0.3  \
+              --weight-decay=0.0 \
+              --num_cores=8  \
+              --max-epoch=%(maxEpoch)d \
+              --log_steps=%(logSteps)d \
+              --train-subset=%(trainSubset)s \
```

Comment:
does `s` work w/ strings?

Reply:
Not sure. I would run `jsonnet` and see if the resulting yaml looks ok

Reply:
Yes, `%s` and `%(var)s` work with strings.